monai/transforms/intensity/array.py

```diff
@@ -1137,3 +1131,141 @@ def __call__(self, img: np.ndarray) -> np.ndarray:
         return np.asarray(
             np.interp(img, reference_control_points_scaled, floating_control_points_scaled), dtype=img.dtype
         )
+
+
+class RandGibbsNoise(RandomizableTransform):
+    """
+    Naturalistic image augmentation via Gibbs artifacts. The transform 
+    randomly applies Gibbs noise to 3D MRI images. Gibbs artifacts
+    are one of the common type of type artifacts appearing in MRI scans.
+    
+    For general information on Gibbs artifacts, please refer to:
+    https://pubs.rsna.org/doi/full/10.1148/rg.313105115
+    https://pubs.rsna.org/doi/full/10.1148/radiographics.22.4.g02jl14949
+
+
+    Args:
+        prob (float): probability of applying the transform.
+        alpha (float): Parametrizes the intensity of the Gibbs noise filter applied. Takes
+            values in [0,1] with alpha = 0 acting as the identity mapping. 
+            If a length-2 list is given as [a,b] then the value of alpha will be 
+            sampled uniformly from the interval [a,b]. 0 <= a <= b <= 1.
+    """
+
+    def __init__(self, prob: float = 0.1, alpha: Union[float, List[float]] = 0.5) -> None:
+
+        if type(alpha) == list:
+            assert alpha[1] <= 1 and alpha[0] >= 0, "alpha must take values in [0,1]."
+            assert alpha[0] <= alpha[1], "When alpha = [a,b] we need a < b."
+            assert len(alpha) == 2, "If alpha is a list, then its length must be 2."
+
+        self.alpha = alpha
+        RandomizableTransform.__init__(self, prob=prob)
+
+    def __call__(self, img: torch.tensor) -> torch.tensor:
+
+        # randomize application and possibly alpha
+        self.randomize()
+
+        if self._do_transform:
+            # apply transform
+            transform = GibbsNoise(self.alpha)
+            img = transform(img)
+        return img
+
+    def randomize(self) -> None:
+        """
+        (1) Set random variable to apply the transform.
+        (2) Get alpha from uniform distribution.
+        """
+        super().randomize(None)
+        if type(self.alpha) == list:
+            self.alpha = self.R.uniform(self.alpha[0], self.alpha[1])
+
+
+class GibbsNoise(Transform):
+    """
+    The transform applies Gibbs noise to 3D MRI images. Gibbs artifacts
+    are one of the common type of type artifacts appearing in MRI scans.
+    
+    For general information on Gibbs artifacts, please refer to:
+    https://pubs.rsna.org/doi/full/10.1148/rg.313105115
+    https://pubs.rsna.org/doi/full/10.1148/radiographics.22.4.g02jl14949
+
+
+    Args:
+        alpha: Parametrizes the intensity of the Gibbs noise filter applied. Takes
+            values in [0,1] with alpha = 0 acting as the identity mapping. 
+    """
+
+    def __init__(self, alpha: Union[float, List[float]] = 0.5) -> None:
+
+        assert alpha <= 1 and alpha >= 0, "alpha must take values in [0,1]."
```

Comment:
if alpha is list?

Reply:
ah, alpha won't be a list here. In which case you need to remove `List[float]` from the constructor type.

Reply:
`GibbsNoise` is fully deterministic so it takes one alpha (a float). I have relegated the option of `alpha = [a,b]` for the random versions `RandGibbsNoise` and `RandGibbsNoised`. In that case the if the user gives the interval [a,b], the value of alpha will be sampled uniformly from [a,b], and passed to the instatiation of `GibbsNoise` in the call method by `transform = GibbsNoise(self.alpha)`.